hazelcast/src/main/java/com/hazelcast/map/impl/query/AggregationResultProcessor.java

```diff
@@ -39,6 +43,7 @@ public AggregationResult populateResult(Query query, long resultLimit, Collectio
 
     @Override
     public AggregationResult populateResult(Query query, long resultLimit) {
-        return new AggregationResult(query.getAggregator());
+        Aggregator resultAggregator = serializationService.toObject(serializationService.toData(query.getAggregator()));
```

Comment:
yeah, it's not super-obvious, but with non-empty result the cloning is inside `AccumulationExecutor`. it would be nice to somehow unify it. anyway, not a subject of this PR.